server.js

```diff
@@ -64,6 +74,48 @@ if (localConfigExists) {
   }
 }
 
+const profilerUrl = `http://${host}:${port}`;
+if (argv.profile) {
+  // Not modifying serverConfig.static because that can roughly triple
+  // webpack cached build time.
+  const profileDir = '/profiles/';
+  const profileFile = path.basename(argv.profile);
+  const prevSetupMiddlewares = serverConfig.setupMiddlewares;
+  serverConfig.setupMiddlewares = (middlewares, devServer) => {
+    if (prevSetupMiddlewares) {
+      middlewares = prevSetupMiddlewares(middlewares, devServer);
+    }
+    devServer.app.use((req, res, next) => {
+      if (
+        req.url.startsWith(profileDir) &&
+        req.url !== `${profileDir}${profileFile}`
+      ) {
+        res.sendStatus(404);
+      } else {
+        next();
+      }
+    });
+    devServer.app.use(
+      profileDir,
+      express.static(path.resolve(path.dirname(argv.profile)))
+    );
+    return middlewares;
+  };
```

Comment:
I think that any reduction in dependency on the (possibly capricious) internal behavior of `webpack-dev-server` is a good thing, and I've integrated the code above. Changes include:

1. The port for the profile server is OS-allocated instead of fixed.
2. The host for the profile server is the same as for the webpack server.
3. CORS handling.
4. The `open` package is now an explicit dependency.

For consistency, the options in `serverConfig.open` are honored. Theoretically, the port for the webpack server can be OS-allocated as well but that should be in a different PR since it might be complicated.